File: frontend/src/types/blog.ts
export interface Blog {
  id: number;
  authorId: number;
  title: string;
  summary: string;
  content: string;
  coverUrl: string | null;
  status: 'DRAFT' | 'PENDING' | 'PUBLISHED' | 'REJECTED';
  viewCount: number;
  likeCount: number;
  createdAt: string;
  updatedAt: string;
  publishedAt: string | null;
}

export interface CreateBlogRequest {
  title: string;
  summary: string;
  content: string;
  coverUrl?: string;
}

export interface UpdateBlogRequest {
  title?: string;
  summary?: string;
  content?: string;
  coverUrl?: string;
}

export interface BlogListResponse {
  content: Blog[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}

export interface BlogTag {
  id: number;
  name: string;
}

export interface BlogComment {
  id: number;
  blogId: number;
  userId: number;
  content: string;
  parentId: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface BlogCommentListResponse {
  content: BlogComment[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}

export interface CreateCommentRequest {
  content: string;
  parentId?: number;
}

export interface LikeStatusResponse {
  hasLiked: boolean;
  likeCount: number;
}

File: services/core-service/src/main/java/com/cptracker/core/entity/Blog.java
package com.cptracker.core.entity;

import jakarta.persistence.*;
import com.cptracker.core.enums.BlogStatus;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "blogs")
public class Blog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "author_id", nullable = false)
    private Long authorId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(length = 500)
    private String summary;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(name = "cover_url", length = 500)
    private String coverUrl;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private BlogStatus status = BlogStatus.DRAFT;

    @Column(name = "view_count")
    private Integer viewCount = 0;

    @Column(name = "like_count")
    private Integer likeCount = 0;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "published_at")
    private LocalDateTime publishedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
